File: golang/src/p20/main.go
package main

import (
  "fmt"
  "math/big"
)

func main() {
  facResult := big.NewInt(1)
  for i := 1; i <= 100; i++ {
    facResult = facResult.Mul(big.NewInt(int64(i)), facResult)
  }

  sum := 0
  for _, c := range facResult.String() {
    sum += int(c - '0')
  }

  fmt.Println(sum)
}

File: cpp/src/p20/main.cc
#include "BigInt.h"
#include <iostream>

int main() {
  BigInt facResult("1");
  const BigInt end("101");
  const BigInt increment("1");
  for (BigInt current("1"); current < end; current.Add(increment)) {
    facResult.Mul(current);
  }

  int sum = 0;
  for (const auto c : facResult.String()) {
    sum += static_cast<int>(c - '0');
  }

  std::cout << sum << "\n";
}
